omero/sysadmins/admins-with-restricted-privileges.txt

```diff
@@ -348,6 +350,10 @@ Client Details:
     # the formerowner from group, just unsets the ownership. This works fine
     # with Create and Edit Group privilege only
     $ bin/omero user leavegroup testgroup --name formerowner  --as-owner
+    # Move a Dataset to a group where the prospective owner is member
+    # and include all annotations on objects
+    # linked to that Dataset even if they do not belong to the same user
+    $ bin/omero chgrp 5 Dataset:51 --include Annotation
     # Transfer ownership to user 55 of the whole hierarchy of the Project 112
     # including the links, as long as all the objects and links in the
     # hierarchy belong to the same user
```

Comment:
>  in RA and RW groups, the annotations on the hierarchy will not be transferred, but stay with the original use

I'm sorry, I'm completely lost here. With Chgrp2 even transferred things stay with the original user. Moving from RA to RA groups where "all the objects and links in the hierarchy belong to the same user" *does* typically transfer annotations unless they are also on data that is being left behind. I'm not sure if "the group" here is source or destination but if the Chgrp2 is going to happen at all then after that point group membership of data owners mostly doesn't matter. For moving annotations the main differences are,

* basic and comment annotations are treated one way, attachments and tags another, and others like map annotation fall in the middle
* it can matter if the annotation is owned by the person who also owns the annotated data
* it matters if the move is *to* a private group

but then there are also special rules like in a RA group if I tag my image with your tag then now you can't move that tag to another group. The behavior is some mix of what seemed not unreasonable and what satisfied the legacy integration tests.

I think mixed-ownership container hierarchies might indeed all move in RW groups and an image owner's own annotations will tend to move with an image. It might be that other users' annotations *also* move if it is a group owner doing the moving.

So, it probably does depend rather on the specifics: in pursuit of https://trello.com/c/8l3d5N5K/93-test-more-annotations-moves you might want to try creating an integration test to probe the particular corner of interest as `AnnotationMoveTest` goes only so far and even `HierarchyMoveAndPermissionsTest.testMoveRois` doesn't cover all these equivalent cases for ROIs.